_additional/fakeClient/lib/Client.js

```diff
@@ -0,0 +1,129 @@
+const request = require('request-promise');
+
+const dataGenerator = require('./dataGenerator');
+const Logger = require('./Logger');
+
+const MESSAGE_DELAY_MIN = 1000;
+const MESSAGE_DELAY_MAX = 5000;
+
+class Client {
+	constructor(address) {
+		this._address = `http://${address}`;
+		this._isRunning = false;
+		this._messageCount = 0;
+		this._nickname = dataGenerator.getNickname();
+		this._logger = new Logger(address);
+		this._messageTimeoutHandler = null;
+		
+		this._logger.write(`New handler created.`);
+	}
+	
+	/**
+	 * Enables fake client, registers its username and starts the message worker.
+	 */
+	run() {
+		this._isRunning = true;
+		
+		this._register()
+			.then(this._runMessageWorker.bind(this))
+			.catch(error => {
+				this._isRunning = false;
+				this._logger.write(error, 'error');
+			});
+	}
+	
+	/**
+	 * Stops message loop and sends http request to '/disconnect' endpoint.
+	 * @returns {Promise.<undefined>}
+	 */
+	disconnect() {
+		this._isRunning = false;
+		clearTimeout(this._messageTimeoutHandler);
+		
+		const params = {
```

Comment:
Imo funkcja `_requestParams(endpoint) {
		return  {
			method: 'POST',
			url: `${this._address}/connect`,
			form: {
				username: this._nickname
			}
		};
}